src/qibolab/instruments/zhinst.py

```diff
@@ -813,75 +818,102 @@ def measure_relax(self, exp, qubits, relaxation_time, acquisition_type):
         elif len(self.sequence_qibo.qd_pulses) != 0:
             play_after = self.play_after_set(self.sequence_qibo.qd_pulses, "drive")
 
-        # Section on the outside loop allows for multiplex
-        with exp.section(uid="sequence_measure", play_after=play_after):
-            for qubit in qubits.values():
-                if qubit.flux_coupler:
-                    continue
-                q = qubit.name  # pylint: disable=C0103
-                if len(self.sequence[f"readout{q}"]) != 0:
-                    for pulse in self.sequence[f"readout{q}"]:
-                        i = 0
-                        pulse.zhpulse.uid += str(i)
+        readout_schedule = defaultdict(list)
+        qubit_readout_schedule = defaultdict(list)
+        iq_angle_readout_schedule = defaultdict(list)
+        for qubit in qubits.values():
+            if qubit.flux_coupler:
+                continue
+            q = qubit.name  # pylint: disable=C0103
+            iq_angle = qubit.iq_angle
+            if len(self.sequence[f"readout{q}"]) != 0:
+                i = 0
+                for pulse in self.sequence[f"readout{q}"]:
+                    readout_schedule[i].append(pulse)
+                    qubit_readout_schedule[i].append(q)
+                    iq_angle_readout_schedule[i].append(iq_angle)
+                    i += 1
 
-                        # Integration weights definition or load from the chip folder
-                        weights_file = (
-                            INSTRUMENTS_DATA_FOLDER
-                            / f"{self.chip}/weights/integration_weights_optimization_qubit_{q}.npy"
+        i = 0
```

Comment:
Also here I would try to avoid the use of the explicit index... but I am not sure how complex is it, eventually it's ok